### ISSUE TITLE
Document usage pattern for template-only components

### ISSUE THREAD

**Issue by SeanHealy33** [simplabs/ember-test-selectors#278]
I noticed an interesting bug with [template only components](https://github.com/emberjs/rfcs/blob/master/text/0278-template-only-components.md) and this package 

Since `ember-test-selectors` attempts to bind the `data-test` attribute to a components wrapping div it fails silently when there is no wrapping div. I found it hard to debug why tests started failing when converting from regular components to template only components.

I've created a repo to highlight the issue [here](https://github.com/SeanHealy33/ember-test)

```js
  test('selector template-only-component exists', async function(assert) {
    assert.ok($('[data-test-template-component]').length)
  });

  test('selector for div-wrapped-component exists', async function(assert) {
    await visit('/');
    assert.ok($('[data-test-div-component]').length)
  });
```

This [acceptance test](https://github.com/SeanHealy33/ember-test/blob/master/tests/acceptance/application-test.js)  will fail because `data-test-template-component` does not exist despite being set on `{{template-only-component}}` [here](https://github.com/SeanHealy33/ember-test/blob/master/app/templates/application.hbs)

I'm not sure what the correct way to mitigate this issue is, since binding to the first element in the template only component could also cause bizarre unexpected behaviour. 

At the very least I think we should update the [Readme.md](https://github.com/simplabs/ember-test-selectors/blob/master/README.md) to highlight the limitation and possibly have `ember-test-selectors` throw an error when it encounters an attempt to add a selector to a template only component to speed up debugging.

**Comment by Turbo87** [comment]
what you describe is essentially impossible currently. since template-only components don't have a root element there is nothing that we can bind the attribute to.

this will however change if you use angle-bracket components:

```template-only-component.hbs
<h1 ...attributes>I do not have a wrapping div</h1>
```

```hbs
<TemplateOnlyComponent data-test-template-component>
```

should result in:

```template-only-component.hbs
<h1 data-test-template-component>I do not have a wrapping div</h1>
```

> At the very least I think we should update the [Readme.md](https://github.com/simplabs/ember-test-selectors/blob/master/README.md) to highlight the limitation

PRs welcome 😉 

> possibly have `ember-test-selectors` throw an error when it encounters an attempt to add a selector to a template only component to speed up debugging.

that is unfortunately not possible, because wherever components are "called" we have no way of knowing if they are template-only or not

**Comment by SeanHealy33** [comment]
@Turbo87 thanks for the quick reply 🙌 


I'll create a PR to update the readme later on today 💪 

> that is unfortunately not possible, because wherever components are "called" we have no way of knowing if they are template-only or not

Interesting, I wasn't sure of the specifics of the internal implementation but I figured I'd mention it as an idea 🙂

**Comment by Turbo87** [comment]
Resolved by #390
